### PR TITLE
ADEN-3483 Sync UAP

### PR DESCRIPTION
**Pull request by drets**
Push "mobile incontent", "mobile prefooter" and "mobile bottom leaderboard" on scroll if there is a UAP campaign in "top leaderboard".
(release after https://github.com/Wikia/app/pull/10790)
/cc @rogatty

### CODE REVIEW

File: front/common/modules/ads.js [hunk 1]
```diff
@@ -103,6 +110,32 @@ class Ads {
...
+				noUapCallback();
+			});
+		} else if (this.uapResult) {
+			uapCallback();
+		} else {
+			noUapCallback();
+		}
+	}
+
+	getBtfSlots() {
```
Comment (fraszczakszymon) [inline front/common/modules/ads.js]: This getter is used only in places where we check whether slot is ATF/BTF. What do you think about moving the logic here and renaming method to `isAboveTheFold`/`isBelowTheFold`?

Comment (fraszczakszymon) [inline front/common/modules/ads.js]: Discussed offline 👍

File: front/common/modules/ads.js [hunk 2]
```diff
@@ -103,6 +110,32 @@ class Ads {
 		});
 	}
 
+	addUapListeners(uapCallback, noUapCallback) {
+		if (!this.uapCalled) {
+			window.addEventListener('wikia.uap', () => {
```
Comment (rogatty) [inline front/common/modules/ads.js]: Do you think it's worth removing the listener after it's not needed anymore?

File: front/main/app/components/ad-slot.js [hunk 3]
```diff
@@ -1,60 +1,104 @@
...
+		noAds: computed({
+			get() {
+				return false;
+			},
+			set(key, value) {
+				return value !== '' && value !== '0';
+			}
+		}),
+
+		didInsertElementOverride: Ember.on('didInsertElement', function () {
```
Comment (rogatty) [inline front/main/app/components/ad-slot.js]: It's not really an override as the original `didInsertElement` is still called. Maybe `onDidInsertElement`?

File: front/main/app/mixins/ads.js [hunk 4]
```diff
@@ -1,14 +1,14 @@
 import Ember from 'ember';
-import AdSlotComponent from '../components/ad-slot';
 import Ads from 'common/modules/ads';
 
 export default Ember.Mixin.create({
 	adsData: {
 		minZerothSectionLength: 700,
 		minPageLength: 2000,
-		mobileTopLeaderBoard: 'MOBILE_TOP_LEADERBOARD',
+		mobileBottomleaderboard: 'MOBILE_BOTTOM_LEADERBOARD',
```
Comment (rogatty) [inline front/main/app/mixins/ads.js]: Can it be consistent with the top one? `mobileBottomLeaderBoard` - one way or another.

File: front/main/app/mixins/ads.js [hunk 5]
```diff
@@ -39,20 +39,27 @@ export default Ember.Mixin.create({
...
+			componentElement.$().insertBefore(element);
+		}
 
-		element[place](view.$());
-		this.adViews.push(view);
+		this.adViews.push(componentElement);
 
-		view.trigger('didInsertElement');
+		componentElement.didInsertElement();
+		componentElement.didInsertElementOverride();
```
Comment (rogatty) [inline front/main/app/mixins/ads.js]: Oh my... If we really have to call it manually (do we?) then I would name it differently, e.g. `onElementManualInsert()`
Right now it's very misleading.

File: front/common/modules/ads.js [hunk 6]
```diff
@@ -103,6 +110,32 @@ class Ads {
 		});
 	}
 
+	addUapListeners(uapCallback, noUapCallback) {
```
Comment (rogatty) [inline front/common/modules/ads.js]: How about `waitForUap` / `waitForUapResponse` / `waitForUapCallback`?

File: front/common/modules/ads.js (hunk 2 above)
Comment (drets) [inline front/common/modules/ads.js]: :+1: It will be handled in separate ticket (ADEN-3542)

File: front/main/app/mixins/ads.js (hunk 4 above)
Comment (fraszczakszymon) [inline front/main/app/mixins/ads.js]: *Leaderboard please

Comment (drets) [inline front/main/app/mixins/ads.js]: <3

**Comment by fraszczakszymon** [comment]
LGTM

**Comment by drets** [comment]
LGTM

**Pull request merged**